src/ga4gh/vrs/normalize.py

```diff
@@ -3,61 +3,170 @@
     if isinstance(allele.location.sequence, models.SequenceReference):
         sequence_reference = allele.location.sequence
         allele.location.sequence = models.IRI(sequence_reference.refgetAccession)
 
-    sequence = SequenceProxy(data_proxy, allele.location.sequence.root)
+    # Get reference sequence and interval
+    ref_seq = SequenceProxy(data_proxy, allele.location.sequence.root)
+    start = _get_allele_location_pos(allele, use_start=True)
+    if start is None:
+        return input_allele
 
-    ival = (allele.location.start, allele.location.end)
+    end = _get_allele_location_pos(allele, use_start=False)
+    if end is None:
+        return input_allele
 
-    _allele_state = allele.state.type
-    _states_with_sequence = ["ReferenceLengthExpression", "LiteralSequenceExpression"]
-    if _allele_state in _states_with_sequence:
+    ival = (start.value, end.value)
+
+    # Get alleles (the sequences to be normalized) for _normalize
+    if allele.state.sequence:
         alleles = (None, allele.state.sequence.root)
-    elif _allele_state == "RepeatedSequenceExpression" and \
-            allele.state.seq_expr.type in _states_with_sequence:
-        alleles = (None, allele.state.seq_expr.sequence.root)
     else:
         alleles = (None, "")
 
-    new_allele = pydantic_copy(allele)
+    # If one of Reference Allele Sequence or Alternate Allele Sequence is empty,
+    # store the length of the non-empty sequence: this is the Repeat Subunit Length
+    len_ref_seq = len(ref_seq[ival[0]: ival[1]])
+    len_alt_seq = len(alleles[1])
+    if not len_ref_seq and len_alt_seq:
+        # Insertion
+        repeat_subunit_len = len_alt_seq
+    elif len_ref_seq and not len_alt_seq:
+        # Deletion
+        repeat_subunit_len = len_ref_seq
+    else:
+        repeat_subunit_len = 0
 
+    new_allele = pydantic_copy(allele)
     try:
-        new_ival, new_alleles = _normalize(sequence,
+        new_ival, new_alleles = _normalize(ref_seq,
                                            ival,
                                            alleles=alleles,
                                            mode=NormalizationMode.EXPAND,
                                            anchor_length=0)
 
-        new_allele.location.start = new_ival[0]
-        new_allele.location.end = new_ival[1]
-
-        if new_allele.state.type in _states_with_sequence:
-            new_allele.state.sequence = models.SequenceString(new_alleles[1])
+        new_allele.location.start = _get_new_allele_location_pos(
+            new_ival[0], start.pos_type
+        )
+        new_allele.location.end = _get_new_allele_location_pos(
+            new_ival[1], end.pos_type
+        )
+        new_ref_seq = ref_seq[new_ival[0]: new_ival[1]]
+
+        if not new_ref_seq:
+            # If the reference sequence is empty this is an unambiguous insertion.
+            # Return a new Allele with the trimmed alternate sequence as a Literal
+            # Sequence Expression
```

Comment:
I'm not sure if `reference sequence` is referring to the original reference sequence or new reference sequence given (possible) new ival